stor/base.py

```diff
@@ -272,6 +273,22 @@ def open(self, *args, **kwargs):
 
         raise NotImplementedError
 
+    def xopen(self, mode='r', *args, **kwargs):
+        """Open a file-like object, transparently handling gzip compression.
+
+        See: :func:`open`
+        """
+        if self.endswith('.gz'):
+            if mode == 'r':
+                mode = 'rb'
+            if mode == 'w':
```

Comment:
I think these codepaths aren't actually tested (that's what is causing CI to fail).

Additionally, you need to pass the original mode to the gzip.GzipFile object, so perhaps you want something like:

```
if mode in ('r', 'rb'): fp_mode='rb'
if mode in ('w', 'wb'): fp_mode = 'wb'
fp = self.open(fp_mode, *args, **kwargs)
gzfp = gzip.GzipFile(mode=mode, fileobj=fp)
```

Additionally, GzipFile's docs say that it doesn't automatically close the underlying file object - which means that if you do something like this:

```
with stor.xopen('s3://unauthed-bucket/myfile.csv.gz') as fp:
    fp.write('somedata')
```
the exception will not bubble up to the user (I believe)